elastic/kibana
# [Security Solution] Add validations to investigation query in markdown editor

### Summary

This PR adds validations for filters in investigation guide query. Currently validations are not sufficient on the form:
- User can save query without filter field, operator or value (https://github.com/elastic/kibana/issues/153092)
- User can save query with values in invalid format (i.e. string in date time field)

Validations added reference checks in unified search `kibana/src/plugins/unified_search/public/filter_bar/filter_editor/lib/filter_editor_utils.ts`

**How to test**
The `Save changes` button should be disabled if:
- Label of form is empty
- Filter is missing field name, operator type and/or value
- Filter value is of incorrect format, possible types and example fields:
   - date (`@timestamp`)
   - string (`host.name`)
   - number (`event.risk_score`). 
   - boolean (`Target.dll.Ext.code_signature.exists`)
   - ip (`host.ip`)
- For `is one of` and `is not one of` operators, it should not allow user to save if any value specified in the array is invalid
- For range operators, it should not allow user to save if `from` or `to` has invalid value

### After


https://github.com/elastic/kibana/assets/18648970/f0424e10-0099-4e02-be00-d8247ab44389



### Checklist
